### PR TITLE
make case list form more lenient

### PR DESCRIPTION
**Pull request by snopoke**
@nickpell

### CODE REVIEW

File: corehq/apps/app_manager/suite_xml/post_process/workflow.py [hunk 1]
```diff
@@ -400,13 +398,10 @@ def get_target_dm(target_form_dm, case_type, module):
...
-            # This error is raised when we can't find a datum that matches the case type we're looking for among
-            # the list of datums that are common between all the forms in the module which implies that not all
-            # the forms have the same case management configuration.
-            raise SuiteValidationError(
-                u"The '{}' module is not properly configured to have a Case List Registration Form. All forms"
-                u" in the module should have the same case management configuration.".format(module.default_name())
-            )
+            # This either means that the source module (with the registration form) requires datums that the
+            # target module (the module which called the reg form).
+            # OR it could mean that not all the forms in the target module have the same case management configuration.
```
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml/post_process/workflow.py]: E501 line too long (119 > 115 characters)

File: corehq/apps/app_manager/tests/test_case_list_form.py [hunk 2]
```diff
@@ -355,6 +355,27 @@ def test_target_module_different_datums(self):
...
+
+        households, edit_household_form = factory.new_basic_module('households', 'household',
+                                                                   case_list_form=register_household_form)
+        factory.form_requires_case(edit_household_form)
+
+        register_member_module, register_member_form = factory.new_advanced_module('new_member', 'member')
+        factory.form_requires_case(register_member_form, 'household')
+        factory.form_opens_case(register_member_form, 'member', is_subcase=True)
+
+        members, edit_member_form = factory.new_basic_module('members', 'member', case_list_form=register_member_form)
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_case_list_form.py]: E501 line too long (118 > 115 characters)

**Comment by snopoke** [comment]
@millerdev any insight on this test failure?
```
======================================================================
ERROR: test_bucket_path (corehq.blobs.tests.test_migratingdb.TestMigratingBlobDB)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/blobs/tests/test_s3db.py", line 181, in test_bucket_path
    self.db.put(StringIO(b"content"), bucket=bucket)
  File "/mnt/commcare-hq-ro/corehq/blobs/migratingdb.py", line 16, in put
    return self.new_db.put(*args, **kw)
  File "/mnt/commcare-hq-ro/corehq/blobs/s3db.py", line 43, in put
    self._s3_bucket(create=True)
  File "/mnt/commcare-hq-ro/corehq/blobs/s3db.py", line 88, in _s3_bucket
    self.db.meta.client.head_bucket(Bucket=self.s3_bucket_name)
  File "/vendor/lib/python2.7/site-packages/botocore/client.py", line 301, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/vendor/lib/python2.7/site-packages/botocore/client.py", line 398, in _make_api_call
    raise ClientError(parsed_response, operation_name)
ClientError: An error occurred (403) when calling the HeadBucket operation: Forbidden
```

**Comment by czue** [comment]
I've been seeing that intermittently. responds well to kicking, but not
sure the root cause.

**Comment by millerdev** [comment]
That's disappointing. Wonder if that [magic incantation to create the riak admin user](https://github.com/dimagi/docker-riak-cs/blob/master/set-admin-keys.sh#L11-L13) is failing intermittently?

**Comment by millerdev** [comment]
I noticed this a lot on travis today, and kicked a lot of builds as a result. Found that I could reproduce fairly reliably (not 100%) with this:

Setup (using [docker-tools branch](https://github.com/dimagi/commcare-hq/pull/12154)):
```sh
./scripts/docker up -d
./scripts/docker bash web ../wait.sh
# localsettings configured to use docker services
```
Reproduce error:
```sh
./scripts/docker restart riakcs && \
./scripts/docker bash web ../wait.sh riakcs && \
./manage.py test corehq.blobs.tests.test_migratingdb:TestMigratingBlobDB.test_bucket_path
```
Adding a sleep seemed to fix it:
```sh
./scripts/docker restart riakcs && \
./scripts/docker bash web ../wait.sh riakcs && \
sleep 5 && \
./manage.py test corehq.blobs.tests.test_migratingdb:TestMigratingBlobDB.test_bucket_path
```
But that's dirty, so I made the [dimagi/riak-cs](https://github.com/dimagi/docker-riak-cs/commit/1d11431bc7c702fa8a89b1f8899ab6cd07be994e) container wait to start the riak-cs service until riak_kv is up. Seems to have fixed it as I can no longer reproduce the error. The new docker image has been pushed, so hopefully we will not see this anymore.

**Comment by snopoke** [comment]
Nice. Thanks

**Comment by snopoke** [comment]
@millerdev I'm still seeing quite a few builds failing with this same error

**Comment by millerdev** [comment]
Updated again. https://github.com/dimagi/docker-riak-cs/commit/bc20569a6139035837ae1e13d5f779fb66ee22fb

**Pull request merged**